Source/WTF/wtf/JSONValues.cpp

```diff
@@ -460,6 +461,38 @@ RefPtr<JSON::Value> buildValue(const CodeUnit* start, const CodeUnit* end, const
 
 } // anonymous namespace
 
+template<typename Visitor> constexpr decltype(auto) Value::visitDerived(Visitor&& visitor)
+{
+    switch (m_type) {
+    case Type::Null:
+    case Type::Boolean:
+    case Type::Double:
+    case Type::Integer:
+    case Type::String:
+        return std::invoke(std::forward<Visitor>(visitor), static_cast<Value&>(*this));
+    case Type::Object:
+        return std::invoke(std::forward<Visitor>(visitor), static_cast<Object&>(*this));
+    case Type::Array:
+        return std::invoke(std::forward<Visitor>(visitor), static_cast<Array&>(*this));
+    }
+    RELEASE_ASSERT_NOT_REACHED();
+}
+
+template<typename Visitor> constexpr decltype(auto) Value::visitDerived(Visitor&& visitor) const
+{
+    return const_cast<Value&>(*this).visitDerived([&](auto& value) {
+        return std::invoke(std::forward<Visitor>(visitor), std::as_const(value));
+    });
+}
+
+void Value::operator delete(Value* value, std::destroying_delete_t)
+{
+    value->visitDerived([](auto& value) {
```

Comment:
NIT: could we maybe use a different name for the lambda parameter `value` so that it's not the same as the outer function parameter `value`?  maybe `derived`?

ditto below as well :)

Reply:
Changed it to `derived`.